lib/document.js

```diff
@@ -2455,13 +2457,23 @@ Document.prototype.$__validate = function(pathsToValidate, options, callback) {
     pathDetails[0].filter((path) => this.isModified(path)) :
     pathDetails[0];
   const skipSchemaValidators = pathDetails[1];
-
   if (typeof pathsToValidate === 'string') {
     pathsToValidate = pathsToValidate.split(' ');
   }
   if (Array.isArray(pathsToValidate)) {
     paths = _handlePathsToValidate(paths, pathsToValidate);
   }
+  else if (pathsToSkip) {
+    // make a new variable that is paths intersection !options.pathsToSkip
+    const valPaths = new Set(paths);
+    for (const path of pathsToSkip) {
+      if (valPaths.has(path)) {
+        valPaths.delete(path);
```

Comment:
I think you missed the point here. The point is to move this logic into `_handlePathsToSkip()`, convert `pathsToSkip` into a set, and use `paths = paths.filter(p => !pathsToSkip.has(p))`